zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -58,4 +155,75 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_founders_address() -> Result<(), Report> {
+        founders_address_for_network(Network::Mainnet)?;
+        founders_address_for_network(Network::Testnet)?;
+
+        Ok(())
+    }
+
+    fn founders_address_for_network(network: Network) -> Result<(), Report> {
+        // First address change after blossom for the 2 networks.
+        // Todo: explain this better and make sure it is working propertly.
+        // after blossom there is still time left for the next change in
+        // founder address but the remaining is calculated with the new formula.
```

Comment:
This doesn't look as though it is using the same logic as the spec. It might be equivalent (I don't know), but I really recommend following the spec closely.

Reply:
@daira question. I am changing this test and i am facing some inconsitency between the testnet and the mainnet. For the entire founder reward period my code is rewarding 48 addresses in the mainnet and 46 in the testnet. Is this possible(and well known) or it is some mistake i am making ?

Reply:
To make this issue more clear i pushed the modified test at https://github.com/ZcashFoundation/zebra/pull/1170/commits/5efae08bc5225e1096ac64e7f52a5ba590ffcf0c and added a comment where the problem is happening at https://github.com/ZcashFoundation/zebra/pull/1170/files#r522501873

Reply:
There are indeed a different number of Founders' Reward periods on mainnet and testnet, because Canopy activated before the halving on testnet. I haven't checked thoroughly that 46 is correct for testnet, but it sounds plausible: there were three extra funding stream periods on testnet before the halving, and if the activation happened partway through an FR period then that means there will be two fewer FR periods.

Reply:
Thanks @daira :+1: